telosnetwork/telos-wallet
# fully functional send page

# Fixes #223 
## also Fixes #174 presented in [this draft PR](https://github.com/telosnetwork/telos-wallet/pull/232)
## also Fixes #253 solved [in this PR](https://github.com/telosnetwork/telos-wallet/pull/255) currently merged

## Description
In order to fulfill the requirement of having a fully functional "send page", three largely independent parts needed to be completed:

- ([#253](https://github.com/telosnetwork/telos-wallet/issues/253)) The implementation of new notification banners for success and error messages.
- ([#174](https://github.com/telosnetwork/telos-wallet/issues/174)) The visual design of the send page, including the HTML and CSS.
- ([#223](https://github.com/telosnetwork/telos-wallet/issues/223)) Adding support for token sending in the new library, which can be further divided into three parts:
  - Retrieving balances for each ERC20 or native token.
  - Obtaining the gas price and estimating the cost of execution.
  - Actual sending of the ERC20 or native token.


## Test scenarios
- goto [this link](https://deploy-preview-257--wallet-staging.netlify.app/)
- login on EVM using an account with at least 1 liquid TLOS and a positive balance of any other ERC20 token
- goto Send Page
  - you should see the send tokens form
  - TLOS should be selected by default
  - the estimated fee should be low but not zero (in TLOS)
  - view token link should not be visible
  - the whole balance should be set in the amount's input.
  - the input should show the amount followed by a semi-transparent symbol and the end
    - the symbol should not affect the input content and should not be editable or even selected
  - the available amount should show the total token balance
  - the fiat balance should be shown in the footer of the input field
- clean the amount field and press the available amount link
  - the whole balance should be set in the amount's input.
- click the fiat balance
  - the footer balance and the input balance should be interchanged
  - the available amount should show the total fiat balance.
  - the input should show the USD symbol and then the input field amount (not editable or selectable)
- change the token
  - the view token contract link should be shown in the token selector footer
  - the available balance should show the total balance for the new token selected
  - if no price is available for that token, then the footers for the amount input are not shown.
  - the estimated fee should be higher than the system token sending gas
- prepare an invalid form (no address, not enough balance, or non-positive amount)
  - the send button should be disabled
- prepare a valid form using any token and press send
  - The Metamask should present the sending confirmation with the correct information, confirm!!
  - If no errors a success notification should be shown for at least 5 sec
    - if you press the see transaction link you should be redirected to Teloscan to see the transaction
  - if an error occurs you should see an error notification with the error info.


## Checklist:
